lib/std/tar.zig

```diff
@@ -243,6 +247,33 @@ pub const IteratorOptions = struct {
 
 /// Iterates over files in tar archive.
 /// `next` returns each file in `reader` tar archive.
+///
+/// Init iterator with tar archive reader and provided buffers:
+///
+///      var file_name_buffer: [std.fs.MAX_PATH_BYTES]u8 = undefined;
+///      var link_name_buffer: [std.fs.MAX_PATH_BYTES]u8 = undefined;
+///
+///      var iter = std.tar.iterator(archive.reader(), .{
+///          .file_name_buffer = &file_name_buffer,
+///          .link_name_buffer = &link_name_buffer,
+///      });
```

Comment:
Don't put code in doc comments. Instead, use a doctest like this:

```zig
test iterator {
    ...
}
```

Then it will show up as example usage like this:

![image](https://github.com/ziglang/zig/assets/106511/5e43a90b-61d1-4ca3-b49b-bb9818ff6301)

This will require the test to be valid code that passes the tests, and therefore cannot bitrot.